tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/FastReconnectOpenWireTest.java

```diff
@@ -0,0 +1,281 @@
+   public void testFastReconnectCreateConsumerNoErrors() throws Exception {
+
+      final ArrayList<Throwable> errors = new ArrayList<>();
+      SimpleString durableQueue = new SimpleString("exampleQueueTwo");
+      this.server.createQueue(new QueueConfiguration(durableQueue).setRoutingType(RoutingType.ANYCAST));
+
+      Queue queue = new ActiveMQQueue(durableQueue.toString());
+
+      final ActiveMQConnectionFactory exFact = new ActiveMQConnectionFactory("failover:(tcp://localhost:61616?closeAsync=false)?startupMaxReconnectAttempts=-1&maxReconnectAttempts=-1&timeout=5000");
+      exFact.setWatchTopicAdvisories(false);
+      exFact.setConnectResponseTimeout(10000);
+      exFact.setClientID("myID");
+
+      RedeliveryPolicy redeliveryPolicy = new RedeliveryPolicy();
+      redeliveryPolicy.setRedeliveryDelay(0);
+      redeliveryPolicy.setMaximumRedeliveries(-1);
+      exFact.setRedeliveryPolicy(redeliveryPolicy);
+
+      publish(1000, durableQueue.toString());
+
+      final AtomicInteger numIterations = new AtomicInteger(NUM_ITERATIONS);
+      ExecutorService executor = Executors.newCachedThreadPool();
+      runAfter(executor::shutdownNow);
+
+      final int concurrent = 2;
+      for (int i = 0; i < concurrent; i++) {
+         executor.execute(() -> {
+            while (numIterations.decrementAndGet() > 0) {
+               try (
+                  Connection conn = exFact.createConnection();
+                  Session consumerConnectionSession = conn.createSession(Session.SESSION_TRANSACTED);
+                  MessageConsumer messageConsumer = consumerConnectionSession.createConsumer(queue)) {
+
+                  messageConsumer.receiveNoWait();
+
+                  if (numIterations.get() % 2 == 0) {
+                     TimeUnit.MILLISECONDS.sleep(30);
+                  } else {
+                     TimeUnit.MILLISECONDS.sleep(50);
+                  }
+
+                  try {
+                     // force a local socket close such that the broker sees an exception on the connection and fails the consumer via serverConsumer close
+                     ((ActiveMQConnection) conn).getTransport().narrow(TcpTransport.class).stop();
+                  } catch (Throwable expected) {
+                  }
+
+               } catch (javax.jms.InvalidClientIDException expected) {
+                  // deliberate clash across concurrent connections
+               } catch (Throwable unexpected) {
+                  unexpected.printStackTrace();
+                  errors.add(unexpected);
+                  numIterations.set(0);
+               }
+            }
+         });
+      }
+
+      executor.shutdown();
+      assertTrue(executor.awaitTermination(60, TimeUnit.SECONDS));
+
+      Wait.assertEquals(0, () -> server.locateQueue(durableQueue).getConsumerCount());
+      assertTrue(errors.isEmpty());
+
+   }
+
+   @Test(timeout = 60_000)
+   public void testFastReconnectCreateConsumerNoErrorsNoClientIdA() throws Exception {
```

Comment:
this test is now a duplicate of the one that comes after, it can safely be deleted.